app/models/note.rb

```diff
@@ -89,24 +91,18 @@ def freshly_closed_until
     closed_at + DEFAULT_FRESHLY_CLOSED_LIMIT
   end
 
-  # Return the note's description, derived from the first comment
-  def description
-    comments.first.body
+  def author_deleted?
+    !author.nil? && author.status == "deleted"
   end
 
-  # Return the note's author object, derived from the first comment
+  # Return the note's author object, unless record is unavailable and
+  # it will be derived from the first comment
   def author
-    comments.first.author
-  end
-
-  # Return the note's author ID, derived from the first comment
-  def author_id
-    comments.first.author_id
-  end
-
-  # Return the note's author IP address, derived from the first comment
-  def author_ip
-    comments.first.author_ip
+    if user_ip.nil? && user_id.nil?
+      all_comments.first.author
+    else
+      association(:author).reader
```

Comment:
Is this going to work?

```suggestion
      super
```

`association` is [not documented](https://github.com/rails/rails/blob/cf6ff17e9a3c6c1139040b519a341f55f0be16cf/activerecord/lib/active_record/associations.rb#L51).

Reply:
Thanks, fixed!